content/en/blog/_posts/2022-10-03-add-userns-alpha/index.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: blog
+title: "Kubernetes 1.25: alpha support for running Pods with user namespaces"
+date: 2022-10-03
+slug: userns-alpha
+---
+
+**Authors:** Rodrigo Campos (Microsoft), Giuseppe Scrivano (Red Hat)
+
+Kubernetes v1.25 introduces the support for user namespaces.
+
+This is a major improvement for running secure workloads in
+Kubernetes.  Each pod will have access only to a limited subset of the
+available UIDs and GIDs on the system, thus adding a new security
+layer to protect from other pods running on the same system.
+
+## How does it work?
+A process running on Linux can use up to 4294967296 different UIDs and
+GIDs.
+
+User namespaces is a Linux feature that allows mapping a set of users
+in the container to different users in the host, thus restricting what
+IDs a process can effectively use.
+Furthermore, the capabilities granted in a new user namespace do not
+apply in the host initial namespaces.
```

Comment:
We don't mention whether UID 0 _inside_ the userns gets any special treatment. Might be worth clarifying?

Reply:
UID 0 inside the userns doesn't get any special treatment and it is treated as any other UID.  I am not sure it is worth mentioning this detail

Reply:
OK, cool. I actually thought UID 0 did get some special treatment - that might be done specially and not a feature of the userns mechanism itself.